lambdas/services/document_deletion_service.py

```diff
@@ -37,21 +40,25 @@ def delete_specific_doc_type(
         nhs_number: str,
         doc_type: Literal[SupportedDocumentTypes.ARF, SupportedDocumentTypes.LG],
     ) -> list[DocumentReference]:
-        results = self.document_service.fetch_available_document_references_by_type(
-            nhs_number, doc_type
-        )
-
-        if not results:
-            return []
-
-        self.document_service.delete_documents(
-            table_name=doc_type.get_dynamodb_table_name(),
-            document_references=results,
-            type_of_delete=str(S3LifecycleTags.SOFT_DELETE.value),
-        )
-
-        logger.info(
-            f"Deleted document of type {doc_type.value}",
-            {"Result": "Successful deletion"},
-        )
-        return results
+        try:
+            results = self.document_service.fetch_available_document_references_by_type(
+                nhs_number, doc_type
+            )
+
+            if not results:
+                return []
+
+            self.document_service.delete_documents(
+                table_name=doc_type.get_dynamodb_table_name(),
+                document_references=results,
+                type_of_delete=str(S3LifecycleTags.SOFT_DELETE.value),
+            )
+
+            logger.info(
+                f"Deleted document of type {doc_type.value}",
+                {"Result": "Successful deletion"},
+            )
+            return results
+        except (ClientError, DynamoServiceException) as e:
+            logger.error(f"Failed to delete documents: {str(e)}")
```

Comment:
Can we add the results part `'"Result": f"Unsuccessful deletion due to {str(e)}"}'` to the log here?